metals/src/main/scala/scala/meta/internal/metals/StacktraceAnalyzer.scala

```diff
@@ -0,0 +1,227 @@
+        java.util.Collections.emptyList()
+    }
+  }
+
+  private def adjustPosition(lineNumber: Int, pos: l.Position): Unit = {
+    pos.setLine(lineNumber)
+    pos.setCharacter(0)
+  }
+
+  private def tryGetLineNumberFromStacktrace(line: String): Try[Int] = {
+    Try(
+      // number is 1-based, but editors are 0-based
+      Integer.valueOf(
+        line.substring(line.indexOf(":") + 1, line.indexOf(")"))
+      ) - 1
+    )
+  }
+
+  private def convert(symbol: String): String = {
+    val components =
+      symbol.split('.').map(_.replace("$", "").replace("<init>", "init")).toList
+    val scope = components.dropRight(1)
+    val method = components.last
+    scope.mkString("/") ++ "." ++ method ++ "()."
+  }
+
+  def stacktraceLenses(content: List[String]): Seq[l.CodeLens] = {
+    (for {
+      (line, row) <- content.zipWithIndex
+      if line.trim.startsWith("at ")
+      location <- getSymbolLocationFromLine(line)
+      range = new l.Range(new l.Position(row, 0), new l.Position(row, 0))
+    } yield makeGotoLocationCodeLens(location, range)).toSeq
+  }
+
+  private def makeGotoLocationCodeLens(
+      location: l.Location,
+      range: l.Range
+  ): l.CodeLens = {
+    new l.CodeLens(
+      range,
+      new l.Command(
+        s"$gotoicon open",
+        ServerCommands.GotoLocationForPosition.id,
+        List[Object](location: Object, java.lang.Boolean.TRUE).asJava
+      ),
+      null
+    )
+  }
+
+  private def analyzeStackTrace(
+      stacktrace: String
+  ): Option[l.ExecuteCommandParams] = {
+    if (commandsInHtmlSupported) {
+      Some(makeHtmlCommandParams(stacktrace))
+    } else {
+      val path = workspace.resolve(Directories.stacktrace)
+      val pathStr = path.toFile.toString
+
+      path.toFile.createNewFile()
+      val fw = new FileWriter(pathStr)
+      try {
+        fw.write(stacktrace)
+      } finally {
+        fw.close()
+      }
+      val pos = new l.Position(0, 0)
+      val range = new l.Range(pos, pos)
+      val location = new l.Location(pathStr, range)
+      Some(makeGotoCommandParams(location))
+    }
+  }
+
+  private def makeGotoCommandParams(
+      location: Location
+  ): l.ExecuteCommandParams = {
+    new l.ExecuteCommandParams(
+      ClientCommands.GotoLocation.id,
+      List[Object](location, java.lang.Boolean.TRUE).asJava
+    )
+  }
+
+  private def getSymbolLocationFromLine(line: String): Option[l.Location] = {
+    val symbol = getSymbolFromLine(line)
+    definitionProvider.fromSymbol(convert(symbol)).asScala.headOption.map {
+      location =>
+        val lineNumberOpt = tryGetLineNumberFromStacktrace(line)
+        lineNumberOpt.foreach { lineNumber =>
+          adjustPosition(lineNumber, location.getRange().getStart())
+          adjustPosition(lineNumber, location.getRange().getEnd())
+        }
+        location
+    }
+  }
+
+  private def getSymbolFromLine(line: String): String = {
+    line.substring(line.indexOf("at ") + 3, line.indexOf("("))
```

Comment:
I don't think you can properly get symbol from stacktrace and the logic will always be flaky. Jumping to the line is enough. For example:
```
Exception in thread "main" java.lang.Exception
	at Main$.$anonfun$main$1(Main.scala:13)
	at Main$.$anonfun$main$1$adapted(Main.scala:10)
	at scala.collection.immutable.Range.foreach(Range.scala:190)
	at Main$.main(Main.scala:10)
	at Main.main(Main.scala)
```

It's not possible to map the full symbol `Main$.$anonfun$main$1` to anything similar in sematicDB. We can only map a part of it, which will make the logic flaky. I would drop going to definition.

Reply:
Although we could do `toplevelFromLine` and just get the toplevel via:
```scala
private def getSymbolFromLine(line: String): String = {
  val toplevel = line.substring(line.indexOf("at ") + 3, line.indexOf("."))
  val descriptor = if(toplevel.endsWith("$")) List(Descriptor.Term) else List(Descriptor.Type)
  BuildTargetClasses.symbolFromClassName(toplevel.stripSuffix("$"), descriptor)
}
```

We might need to move the method from BuildTargetClasses if it makes sense. It's in the class, so some modification will be needed.

Reply:
okay dropping